pandas/io/tests/test_pytables.py

```diff
@@ -4733,6 +4733,79 @@ def test_invalid_complib(self):
         with ensure_clean_path(self.path) as path:
             self.assertRaises(ValueError, df.to_hdf, path, 'df', complib='blosc:zlib')
 
+    # GH10447
+    def test_complex(self):
+        df = DataFrame(np.random.rand(4, 5).astype(np.complex64),
+                       index=list('abcd'),
+                       columns=list('ABCDE'))
+        with ensure_clean_path(self.path) as path:
```

Comment:
An example?  I am worried that the block from PyTables

```python
    if dtype.kind == 'c':
        raise TypeError("complex columns can not be indexed")
```

might be a deal breaker for doing things like `store.select`